### PR TITLE
Fixed AllowGlobalScope update state issue

### PR DESCRIPTION
**Pull request by Shpakh-MSFT**
My previous change to add support for AllowAccessToGlobals property wasn't complete. It failed in roundtripping validation. This change moved the IR property to ComponentManifest instead.

### CODE REVIEW

File: src/PAModel/EditorState/ControlState.cs [hunk 1]
```diff
@@ -29,6 +29,8 @@ internal class ControlState
         // even though this can be re-derived from the existence of DynamicProperties
         public bool? HasDynamicProperties { get; set; }
 
+        public bool? AllowAccessToGlobals { get; set; }
```
Comment (lesaltzm) [inline src/PAModel/EditorState/ControlState.cs:32]: Is this used from ControlState, or was this testing?

File: src/PAModel/IR/IRStateHelpers.cs [hunk 2]
```diff
@@ -336,12 +336,13 @@ private static (ControlInfoJson.Item item, int index) CombineIRAndState(BlockNod
                     StyleName = state.StyleName,
                     ExtensionData = state.ExtensionData,
                     IsGroupControl = state.IsGroupControl,
-                    GroupedControlsKey = state.GroupedControlsKey
+                    GroupedControlsKey = state.GroupedControlsKey,
+                    AllowAccessToGlobals = (state.IsComponentDefinition ?? false) ? (templateState?.ComponentManifest?.AllowAccessToGlobals ?? true) : true,
```
Comment (lesaltzm) [inline src/PAModel/IR/IRStateHelpers.cs:340]: this default to true has broken our existing roundtrip tests, looks like most apps don't have this set at all.

File: src/PAModel/EditorState/ControlState.cs (hunk 1 above)
Comment (Shpakh-MSFT) [reply src/PAModel/EditorState/ControlState.cs:32]: Not needed. Good catch. Thanks.

Comment (Shpakh-MSFT) [reply src/PAModel/EditorState/ControlState.cs:32]: This is now used.

**Review by lesaltzm — APPROVED** [review]
🚢

**Pull request merged**